src/base/cub_ddl_log.c

```diff
@@ -0,0 +1,1206 @@
+
+  name_tmp = strrchr (file_full_path, PATH_SEPARATOR);
+
+  if (name_tmp == NULL)
+    {
+      name_tmp = file_full_path;
+    }
+  else
+    {
+      name_tmp++;
+    }
+
+  retval = snprintf (ddl_audit_handle->schema_file, buf_size, "%s_%s_%d", name_tmp, time, getpid ());
+  if (retval < 0)
+    {
+      assert (false);
+      ddl_audit_handle->schema_file[0] = '\0';
+      return retval;
+    }
+
+  retval =
+    snprintf (dest_path, buf_size, "%s/%s/%s_%s_%d", env_root, DDL_LOG_LOADDB_FILE_PATH, name_tmp, time, getpid ());
+
+  if (retval < 0)
+    {
+      assert (false);
+      dest_path[0] = '\0';
+      return retval;
+    }
+
+  retval = cub_create_dir_log (dest_path);
+
+  return retval;
+}
+
+static int
+cub_make_ddl_log_filename (char *filename_buf, size_t buf_size, T_APP_NAME app_name)
+{
+  const char *env_root = NULL;
+  int retval = 0;
+
+  assert (filename_buf != NULL);
+
+  env_root = envvar_root ();
+
+  if (app_name == APP_NAME_CAS)
+    {
+      retval =
+	snprintf (filename_buf, buf_size, "%s/%s/%s_%d_ddl.log", env_root, DDL_LOG_PATH, ddl_audit_handle->br_name,
+		  (ddl_audit_handle->br_index + 1));
+    }
+  else if (app_name == APP_NAME_CSQL || app_name == APP_NAME_LOADDB)
+    {
+      retval =
+	snprintf (filename_buf, buf_size, "%s/%s/%s_%s_ddl.log", env_root, DDL_LOG_PATH, cub_get_app_name (app_name),
+		  ddl_audit_handle->db_name);
+    }
+  else
+    {
+      filename_buf[0] = '\0';
+      return -1;
+    }
+
+  if (retval < 0)
+    {
+      assert (false);
+      filename_buf[0] = '\0';
+    }
+
+  return retval;
+}
+
+static FILE *
+cub_ddl_log_open (T_APP_NAME app_name)
+{
+  FILE *fp = NULL;
+  char *tpath = NULL;
+  int len;
+
+  fprintf (stderr, "=========== cub_ddl_log_open : %s \n", cub_get_app_name (app_name));
+
+  len = cub_make_ddl_log_filename (ddl_audit_handle->log_filepath, PATH_MAX, app_name);
+
+  if (ddl_audit_handle->log_filepath[0] == '\0' || len < 0)
+    {
+      goto file_error;
+    }
+
+  if (cub_create_dir_log (ddl_audit_handle->log_filepath) < 0)
+    {
+      goto file_error;
+    }
+
+  /* note: in "a+" mode, output is always appended */
+  fp = cub_fopen_and_lock (ddl_audit_handle->log_filepath, "a+");
+
+  return fp;
+
+file_error:
+  return NULL;
+}
+
+void
+cub_ddl_log_write_end ()
+{
+  if (ddl_audit_handle == NULL)
+    {
+      return;
+    }
+
+  if (prm_get_bool_value (PRM_ID_DDL_AUDIT_LOG) == FALSE)
+    {
+      goto ddl_log_free;
+    }
+
+  if (ddl_audit_handle->app_name == APP_NAME_LOADDB)
+    {
+      if (ddl_audit_handle->loaddb_file_type != LOADDB_FILE_TYPE_SCHEMA
+	  && ddl_audit_handle->loaddb_file_type != LOADDB_FILE_TYPE_INDEX
+	  && ddl_audit_handle->loaddb_file_type != LOADDB_FILE_TYPE_TRIGGER)
+	{
+	  goto ddl_log_free;
+	}
+    }
+  else
+    {
+      if (cub_is_ddl_type (ddl_audit_handle->stmt_type) == FALSE)
+	{
+	  goto ddl_log_free;
+	}
+    }
+  if (ddl_audit_handle->execute_type & DDL_LOG_RUN_EXECUTE_FUNC)
```

Comment:
For readability, I prefer
`if (ddl_audit_handle->execute_type == DDL_LOG_RUN_EXECUTE_FUNC)`

Reply:
Thank you, i will change code as your recommend.